### PR TITLE
JWWEB-232 SettingsCard implementation

### PR DESCRIPTION
**Pull request by Creepypoke**
SettingsCard component implementation.

### CODE REVIEW

File: src/components/SettingsCard/SettingsCard.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  iconName?: string,
+  iconColor?: 'white' | 'blue' | 'gray' | 'sky',
+  iconSize?: 'small' | 'medium' | 'large' | 'xlarge',
+}
+
+SettingsCard.defaultProps = {
+  iconName: 'setting',
+  iconColor: 'gray',
+  iconSize: 'medium',
+}
```
Comment (IViol) [inline src/components/SettingsCard/SettingsCard.js]: if icon props could be excluded from component props, why we set some default values for them?
all of them could be null.
And before using of them, we should check `iconName && iconColor && iconSize && <...>`

File: src/components/SettingsCard/SettingsCard.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        />
+      </div>}
+      <div className='summary'>
+        <div className='title'>
+          <JText value={title} color='dark' weight='bolder' size='header' />
+        </div>
+        {description &&
+        <div className='description'>
+          <span className='text'>{description}</span>
+        </div>}
```
Comment (IViol) [inline src/components/SettingsCard/SettingsCard.js]: description should be wrapped in JText

File: src/components/SettingsCard/settingsCard.scss [hunk 3]
```diff
@@ -0,0 +1,50 @@
+.settings-card {
+  all: initial;
+
+  @include box;
+  @include font;
+  @include flex-column;
+  @include hover-animation;
+
```
Comment (IViol) [inline src/components/SettingsCard/settingsCard.scss]: you don't need to include font mixin if description will be wrapped by JText

File: stories/components/SettingsCrad.stories.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+          max-width: 1020px;
+          margin: 0 auto;
+          display: flex;
+          flex-wrap: wrap;
+        }
+        .settings-grid > .box {
+          width: 33.333%;
+          padding: 0 10px 20px 10px;
+        }`}
+      </style>
```
Comment (IViol) [inline stories/components/SettingsCrad.stories.js]: better to write css in `stories/styles.css`

File: src/components/SettingsCard/SettingsCard.js (hunk 1 above)
Comment (IViol) [reply src/components/SettingsCard/SettingsCard.js]: but if we don't have setting cards without icons, we should make these props required

File: src/components/SettingsCard/SettingsCard.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import { JIcon, JText } from 'components/base'
+
+type Props = {
+  title: string,
+  description: string,
+  path: string,
+  iconName?: string,
+  iconColor?: 'white' | 'blue' | 'gray' | 'sky',
+  iconSize?: 'small' | 'medium' | 'large' | 'xlarge',
+}
```
[the remark this answers is not in the document]
Comment (flexxnn) [reply src/components/SettingsCard/SettingsCard.js]: For better type checking it is good to use strict operator "|", and make props read only
`type Props = {|
  +title: string,
  +description: string,
  +path: string,
  +iconName: string,
  +iconColor: 'white' | 'blue' | 'gray' | 'sky',
  +iconSize: 'small' | 'medium' | 'large' | 'xlarge',
|}`

File: src/components/SettingsCard/SettingsCard.js (hunk 1 above)
Comment (flexxnn) [reply src/components/SettingsCard/SettingsCard.js]: I think, here the `size` is always medium. Isn't it?

Comment (Creepypoke) [reply src/components/SettingsCard/SettingsCard.js]: Both of you are right: `iconName` is should be required, and `iconSize` always is should has `medium` value, only color of icon has options.

**Pull request closed without merging**